src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1684,6 +1689,44 @@ TypeSymbol getRightResultType(TypeSymbol leftType, TypeSymbol rightType)
             }
         }
 
+        /// <summary>
+        /// Return top-level nullability for the expression. This method should be called on a limited
+        /// set of expressions only. It should not be called on expressions tracked by flow analysis
+        /// other than <see cref="BoundKind.ExpressionWithNullability"/> which is an expression
+        /// specifically created in NullableWalker to represent the flow analysis state.
+        /// </summary>
+        private static bool? GetIsNullable(BoundExpression expr)
+        {
+            switch (expr.Kind)
+            {
+                case BoundKind.DefaultExpression:
+                case BoundKind.Literal:
+                    {
+                        var constant = expr.ConstantValue;
+                        if (constant != null)
+                        {
+                            if (constant.IsNull)
+                            {
+                                return true;
+                            }
+                            if (expr.Type?.IsReferenceType == true)
+                            {
+                                return false;
+                            }
+                        }
+                        return null;
+                    }
+                case BoundKind.ExpressionWithNullability:
+                    return ((BoundExpressionWithNullability)expr).IsNullable;
+                case BoundKind.MethodGroup:
+                case BoundKind.UnboundLambda:
+                    return null;
+                default:
+                    Debug.Assert(false); // unexpected value
```

Comment:
>Debug.Assert [](start = 20, length = 12)

Consider throwing UnexpectedException

Reply:
We can add that later perhaps. For now, we probably shouldn't crash the process since the worst outcome of missing a case here is missing warnings.

---
In reply to: [210390086](https://github.com/dotnet/roslyn/pull/29317#discussion_r210390086) [](ancestors = 210390086)